fw/stat.h

```diff
@@ -0,0 +1,111 @@
+/**
+ *		Tempesta kernel library
+ *
+ * Copyright (C) 2016-2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#ifndef __TFW_STAT_H__
+#define __TFW_STAT_H__
+
+#include "http.h"
+
+/* Shared structures used for statistics. */
+
+/*
+ * Structures for health monitoring statistics accountings in procfs.
+ *
+ * @code	- HTTP code for which the statistics is collected.
+ * @sum		- Amount of responses for some limited period.
+ * @total	- Total amount of responses for the overall period.
+ */
+typedef struct {
+	int		code;
+	unsigned int	sum;
+	u64		total;
+} TfwHMCodeStats;
+
+/*
+ * @rtime	- time until next server health checking (can be unused if not
+ *		  required);
+ * @ccnt	- count of @rsums elements;
+ * @rsums	- array of counters for separate HTTP codes;
+ */
```

Comment:
I know this isn't your doing, but why not make it a bit better? Below are thoughts on both better naming and better comments.

Here it's baffling what `rtime` stands for. It it's the `remaining time`, then it's definitely not telling at all. Perhaps, `tmcheck` - check time? `Time until next server health check`. It's unclear what it's meant that it can be unused if these stats are collected unconditionally.

Then, `rsums` could be just `stats` or `cstats` - `Array of stats per HTTP status code`, and `ccnt` -> `stats_sz`  (or `statsz`) - `Size of stats array` as per convention throughout the code.

Also, you can perhaps avoid multiplication to `HZ` every time and keep timers in jiffies as it's done in other parts of APM. Then, `tmcheck` would become `jtmcheck`.